tools/generate_hotkeys.py

```diff
@@ -21,25 +21,26 @@ def generate_hotkeys_file() -> None:
     """
     Generate hotkeys.md in docs folder based on help text description and shortcut key combinations in config/keys.py file
     """
-    categories = read_help_categories()
+    hotkeys_file_string = get_hotkeys_file_string()
+    compare_hotkeys_file(hotkeys_file_string)
     with open(OUTPUT_FILE, "w") as mdFile:
-        mdFile.write(
-            f"<!--- Generated automatically by tools/{SCRIPT_NAME} -->\n"
-            "<!--- Do not modify -->\n\n# Hot Keys\n"
-        )
-        for action in HELP_CATEGORIES.keys():
-            mdFile.write(
-                f"## {HELP_CATEGORIES[action]}\n"
-                "|Command|Key Combination|\n"
-                "| :--- | :---: |\n"
-            )
-            for help_text, key_combinations_list in categories[action]:
-                various_key_combinations = various_key_combination(
-                    key_combinations_list
-                )
-                mdFile.write(f"|{help_text}|{various_key_combinations}|\n")
-            mdFile.write("\n")
-    print(f"Hot Keys list saved in {OUTPUT_FILE}")
+        mdFile.write(hotkeys_file_string)
+        print(f"Hot Keys list saved in {OUTPUT_FILE}")
+
+
+def get_hotkeys_file_string() -> str:
+    """
+    Get existing hotkeys.md in docs folder as string based on help text description and shortcut key combinations in config/keys.py file
+    """
+    categories = read_help_categories()
+    hotkeys_file_string = f"<!--- Generated automatically by tools/{SCRIPT_NAME} -->\n<!--- Do not modify -->\n\n# Hot Keys\n"
+    for action in HELP_CATEGORIES.keys():
+        hotkeys_file_string += f"## {HELP_CATEGORIES[action]}\n|Command|Key Combination|\n| :--- | :---: |\n"
```

Comment:
Here as well, we can keep the original style of:
```python
hotkeys_file_string += (
    f"## {HELP_CATEGORIES[action]}\n"
    "|Command|Key Combination|\n"
    "| :--- | :---: |\n"
)
```
which I'm hoping would improve the diffs, if `categories = read_help_categories()` was inserted inside `with open` in the previous commit.

Reply:
Earlier also listing ``categories`` into dict was outside ``with open``. So I think diff is clearer if ``categories = read_help_categories`` is outside ``with open``

Reply:
Here, I meant all such multi-line `write()`'s. You missed atleast one at the top.
Re categories, I meant change the position in the previous commit. Try with an without this and see the difference.